src/Arachne.h

```diff
@@ -505,16 +514,21 @@ random(void) {
   */
 template<typename _Callable, typename... _Args>
 ThreadId
-createThreadOnCore(uint32_t virtualCoreId, _Callable&& __f, _Args&&... __args) {
-    if (virtualCoreId >= numActiveCores) {
-        ARACHNE_LOG(VERBOSE, "createThread failure, virtualCoreId = %u, "
-                   "numActiveCores = %d\n", virtualCoreId,
+createThreadOnCore(threadClass_t threadClass, uint32_t coreId, _Callable&& __f, _Args&&... __args) {
+
+    threadCoreMapEntry* entry = corePolicy->getThreadCoreMapEntry(threadClass);
+    bool isLegalCoreId = false;
+    for (uint32_t i = 0; i < entry->numFilled; i++) {
+      if (entry->map[i] == (int) coreId)
```

Comment:
I'm not sure what the purpose is to use both coreId and threadClass. Let's discuss.